### PR TITLE
Sign Up

### PR DESCRIPTION
**Pull request by ranuccimariano**
## Summary

- Created the `/users` endpoint to sign_up users.
- Validated fields:
    - Email
    - Password
    - First name
    - Last name
- Added bcrypt to hash passwords. Password must be alphanumeric and must contain 8 or more characters.
- Email must belong to wolox domain. Ex: 'mariano.ranucci@wolox.com.ar'

## Trello Card

https://trello.com/c/PEaW2PU7

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const { userCreationError } = error;
+const saltRounds = 10;
+
+const validateParams = (email, password) =>
+  new Promise((resolve, reject) => {
+    if (isAlphanumeric(password) && password.length >= 8 && email.split('@')[1] === 'wolox.com.ar') {
+      resolve('Valid Params');
+    } else {
+      reject(Error('Invalid params!'));
```
Comment (molinajulian) [inline app/controllers/users.js]: Throw our error

File: app/controllers/users.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  return validateParams(email, password)
+    .then(
+      bcrypt
+        .hash(password, saltRounds)
+        .then(hash => {
+          User.create({ firstName, lastName, email, password: hash });
+          logger.info(`Usuario creado para: ${firstName} ${lastName}`);
+        })
+        .catch(err => {
+          logger.error(util.inspect(err));
```
Comment (molinajulian) [inline app/controllers/users.js]: If you catch here, The cach in line 37, it will catch too

File: app/routes.js [hunk 3]
```diff
@@ -1,8 +1,10 @@
 const { healthCheck } = require('./controllers/healthCheck');
 const { getAlbums, getAlbumPhotos } = require('./controllers/albums');
+const { postUser } = require('./controllers/users');
```
Comment (molinajulian) [inline app/routes.js]: Maybe `createUser` ?

File: app/controllers/users.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+const isAlphanumeric = require('is-alphanumeric');
+
+const util = require('util');
+
+const User = require('../models').users;
+const logger = require('../logger');
+const error = require('../errors');
+
+const { userCreationError } = error;
+const saltRounds = 10;
```
[the remark this answers is not in the document]
Comment (ranuccimariano) [reply app/controllers/users.js]: Done!

File: app/controllers/users.js (hunk 1 above)
Comment (ranuccimariano) [reply app/controllers/users.js]: Done

File: app/controllers/users.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+const validateParams = (email, password) =>
+  new Promise((resolve, reject) => {
+    if (isAlphanumeric(password) && password.length >= 8 && email.split('@')[1] === 'wolox.com.ar') {
+      resolve('Valid Params');
+    } else {
+      reject(userCreationError('Invalid params!'));
+    }
+  });
+
+const hashPassword = password => bcrypt.hash(password, saltRounds);
```
Comment (molinajulian) [inline app/controllers/users.js]: Move `bcrypt` to a service

**Pull request merged**